src/xopen/__init__.py

```diff
@@ -1039,15 +1044,45 @@ def _detect_format_from_extension(filename: Union[str, bytes]) -> Optional[str]:
     return None
 
 
+@overload
+def xopen(
+    filename: Union[str, bytes, os.PathLike],
```

Comment:
What I usually do:
```Python3
FilePath = Union[str, bytes, os.Pathlike]
```
And then use FiePath everywhere for convenience, but YMMV.

Reply:
Done